### ISSUE TITLE
Fix Version for all npm dependencies

### ISSUE THREAD

**Issue by bruderol** [scenarioo/scenarioo#689]
Currently we use versions like 2.0.x in package.json

To me this is not acceptable - because what can go wrong will go wrong! Murphies Law!

Some reasons for this:
* Updating dependencies should always be a well done (at least partially manual process), whcih involves some testing and verifying whether it is ok. 
* builds should always be 100% reproducable! 
* There could always be an inconsistency in compatibility even just for minor versions or even a blocker bug / issue for us that was introduced in a minor version (happened already a lot in the past). 
* I do not want that somebody gets blocked because of such issues, when he has some hours to work on scenarioo. 
* And I do not want that even such a minor change could block us on the day when we want to releaese (already happened once)

package-lock does not solve this problem for direct dependencies as you can see here:
https://stackoverflow.com/questions/45022048/why-does-npm-install-rewrite-package-lock-json

That is why I adjusted docu here (with some further explanation, why this is important):
https://github.com/scenarioo/scenarioo/commit/3175e7cc3b5a0103a6490cea27617448dd775ee8
(see bottom!)

So please let's use fix versions from now on, and then do the updates manually from time to time.

**Acceptance criteria**
[] all dependencies in package.json are changed to use full versions like "2.0.2" (never "2.0,x)
[] the decission is communicated to all in the team
[] the documentation (see link above is adjusted accordingly - more or less done already)
[] The manual update process is docuemented better and should at least include the following steps:
     * going at least through a feature branch to verify on CI that everything is still working ok.
     * quickly verify on windows and mac (!!) whether the npm build and e2e testing locally still works! (with the help of feature branches we can easily verify such things before merging)
     * optimaly: do some manual testing quickly on the feature branch demo before merging

**Comment by bruderol** [comment]
@adiherzog @danielsuter @mhanggi @eggerschwiler @cgrossde feedback appreciated. I hope you all agree?

**Comment by adiherzog** [comment]
Agree. Unfortunately we can't use https://greenkeeper.io because they expect the package.json in the root folder. So we somehow have to make sure that we regularly update our dependencies. The more we do it, the easier.

**Comment by bruderol** [comment]
Yes, but we could easily use something like https://www.npmjs.com/package/npm-check-updates

On the other hand: I think doing the update manually from time to time (e.g. once after each release and once before we create a release candidate) should be sufficient? Important is just that we have a reliable way of updating our dependencies and that this is documented.

**Comment by cgrossde** [comment]
My input on this:
- We could use ‘yarn’ to get the desired behavior.
- There is ‘npm ci’ now which uses exact versions from package-lock: https://docs.npmjs.com/cli/ci
- ‘npm outdated’ or ‘yarn outdated’ offer a nice overview which packages have newer versions, no need for ncu anymore ;)

Am 21. Mai 2018, 00:18 +0200 schrieb Rolf Bruderer <notifications@github.com>:

Yes, but we could easily use something like https://www.npmjs.com/package/npm-check-updates

On the other hand: I think doing the update manually from time to time (e.g. once after each release and once before we create a release candidate) should be sufficient? Important is just that we have a reliable way of updating our dependencies and that this is documented.

**Comment by bruderol** [comment]
put it to next release - maybe currently not that urgent - as long as we not run into problems - but we should really tackle that as part of the angular upgrade thing.
